mysql-test/suite/versioning/t/foreign.test

```diff
@@ -192,6 +192,28 @@ select *, current_row(sys_end) as current_row from child for system_time all ord
 drop table child;
 drop table parent;
 
+--echo # ignore history rows on ALTER
+create table parent(
+  id int unique key
+);
+
+--replace_result $sys_datatype_expl SYS_DATATYPE
+eval create table child(
+  parent_id int,
+  sys_start $sys_datatype_expl as row start invisible,
+  sys_end $sys_datatype_expl as row end invisible,
+  period for system_time(sys_start, sys_end)
+) with system versioning;
+
+insert into child values (1);
+--error ER_NO_REFERENCED_ROW_2, ER_NO_REFERENCED_ROW_2
```

Comment:
Why did you suppress the error message?

Reply:
The easiest way to ignore temporary table name.

Reply:
Generally, it's better to do, like elsewhere,
```
--replace_regex /#sql-[0-9a-f_]*/#sql-temporary/
```
this would not have caused any questions.

But in this case, still, you're using `s->error_table_name()` (not only in this patch in other places too). Is it not enough?

Reply:
Not for this case. This one was a bit hard to find out. I'll recheck.

Reply:
This task is complex enough to create separate issue. Created [MDEV-16301](https://jira.mariadb.org/browse/MDEV-16301). Meanwhile updated to `--replace_regex` as you're suggesting.